lib/computer.ex

```diff
@@ -2,31 +2,28 @@ defmodule Computer do
   defstruct mark: nil
 
   def move(game) do
-    pick_move(game)
+    Game.mark_spot(game, get_best_move(game))
   end
 
-  def pick_move(game) do
-    available_spaces_number = game.board |> Board.available_spaces_number
-    cond do
-      available_spaces_number >= 8 and game.size == 3 ->
-        move = Enum.random(0..8)
-        if Board.is_available?(game.board, move), do: Game.mark_spot(game, move), else: move(game) 
-      available_spaces_number >= 13 and game.size == 4 ->
-        move = Enum.random(0..15)
-        if Board.is_available?(game.board, move), do: Game.mark_spot(game, move), else: move(game)
-      true ->
-        Game.mark_spot(game, get_best_move(game, game.current_player))
-    end    
+  def get_best_move(game, over \\ false, scores \\ [], depth \\ 0)
+  def get_best_move(game, true, _scores, depth), do: hueristic_score(game, depth)    
+  def get_best_move(game, false, _scores, depth) do
+    scores = 
+      Enum.map(game |> Board.available_spaces, fn(space) ->
+        game = Game.mark_spot(game, space) 
+        score = if depth < 4, do: get_best_move(game, Game.over?(game), [], depth + 1)
+        {space, score}
+      end)
+    %player{} = game.current_player
+    if depth == 0, do: best_move(scores), else: minimax_score(player, scores)
   end
```

Comment:
So, this function:

* Marks a spot
* Performs move analysis
* Ranks moves

Your work here (and also Elixir is probably helping :) ) is pretty nice, it reads very succinctly.